lib/luaxtable.c

```diff
@@ -81,7 +81,9 @@ static int luaxtable_pushparams(lua_State *L, const struct xt_action_param *par,
 		pr_err("could not find skb\n");
 		return -1;
 	}
-	luadata_reset(data, skb->data, skb_headlen(skb), opt);
+	if (skb_linearize(skb) != 0)
+          return -1;
+        luadata_reset(data, skb->data, skb->len, opt);
```

Comment:
```suggestion
	if (unlikely(data == NULL || skb_linearize(skb) != 0)) {
		pr_err("could not get skb\n");
		return -1;
	}
        luadata_reset(data, skb->data, skb->len, opt);
```

I would simplify the patch in this way.. please notice that we would need to remove the [lines above](https://github.com/luainkernel/lunatik/pull/155/files#diff-292f6a11c6f52f91544b0b47343e04b5d6516316ca5a075844659348899709ffR80-R83) as well.. what do you think?